### PR TITLE
docker_compose: fix #12

### LINKED ISSUE #12: docker_compose: Cannot pull images without starting and stopping stopped containers
##### SUMMARY

As @slavik57  wrote 

> Currently there is no way to pull the images without starting the containers.
> When running with state:absent and pull:true it won't pull the images.
> When running with state:present and pull:true and stopped:yes it will pull the images, start the containers and then stop them.
> This may cause some unwanted behaviour such as volume creation if there is a local volume mapping.

From: https://github.com/ansible/ansible/issues/31924

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
docker_compose

##### ANSIBLE VERSION

2.x

##### OS / ENVIRONMENT
All

See previous discussion at https://github.com/ansible/ansible/issues/31924

It is my (@abeluck) opinion that we should maintain backwards compatibility. And so  `state: present` and `stopped: true` and `pull: true` should effectively dun docker-compose up --no-start --pull. And if any services are already stopped, then stop them.

### PR DESCRIPTION
**Pull request by tonal**
##### SUMMARY
Fixes #12.

##### ISSUE TYPE

- Bugfix Pull Request

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Review by felixfontein — COMMENTED** [review]
Can you please add a [changelog fragment](https://docs.ansible.com/ansible/devel/community/development_process.html#changelogs-how-to)?

File: changelogs/fragments/12-correct_pull_wo_starting.yaml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+bugfixes:
+- docker-compose - fixed not pull if ``state: present`` and ``stopped: true``
```
Comment (felixfontein) [inline changelogs/fragments/12-correct_pull_wo_starting.yaml:2]:
```suggestion
- "docker-compose - fixed not pull if ``state: present`` and ``stopped: true`` (https://github.com/ansible-collections/community.docker/issues/12, https://github.com/ansible-collections/community.docker/pull/119)."
```

**Review by felixfontein — COMMENTED** [review]
I think for this to work properly, `result['changed'] = stop_output['changed']` in line 798 needs to be changed to `result['changed'] |= stop_output['changed']`.

**Comment by felixfontein** [comment]
> I think for this to work properly, `result['changed'] = stop_output['changed']` in line 798 needs to be changed to `result['changed'] |= stop_output['changed']`.

@tonal what do you think about this?

File: changelogs/fragments/12-correct_pull_wo_starting.yaml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+bugfixes:
+- "docker-compose - fixed not pull if ``state: present`` and ``stopped: true`` (https://github.com/ansible-collections/community.docker/issues/12, https://github.com/ansible-collections/community.docker/pull/119)."
```
Comment (aminvakil) [inline changelogs/fragments/12-correct_pull_wo_starting.yaml:2]: I don't feel like my suggestion is well yet, please improve it any way you fit.
```suggestion
- "docker-compose - fix not pulling when ``state: present`` and ``stopped: true`` (https://github.com/ansible-collections/community.docker/issues/12, https://github.com/ansible-collections/community.docker/pull/119)."
```

**Review by aminvakil — APPROVED** [review]

**Review by aminvakil — COMMENTED** [review]
I thought @felixfontein has commited this change himself:
> I think for this to work properly, `result['changed'] = stop_output['changed']` in line 798 needs to be changed to `result['changed'] |= stop_output['changed']`.

Sorry, I think this has to be changed before merging.

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
@tonal thanks for fixing this!
@aminvakil thanks for reviewing!

**Pull request merged**